src/deye_modbus.py

```diff
@@ -211,3 +208,51 @@ def __parse_response_error_code(self, frame: bytes) -> None:
             self.__log.error("Logger Serial Number does not match. Check your configuration file.")
         else:
             self.__log.error("Unknown response error code. Error frame: %s", error_frame.hex())
+
+    def crc16(self, data: bytes) -> bytes:
```

Comment:
Why not to simply add modbus-crc as a project dependency instead of copying over the code?

Reply:
It is very little code and I wanted to avoid a new dependency. You are welcome to change it, if you want to!

Reply:
> It is very little code and I wanted to avoid a new dependency.

I see no problem with that. You basically replace `libscrc` with way more lightweight library. 

> You are welcome to change it, if you want to!

Thanks, yet I doubt if I'm allowed to push to your branch. And to be honest, I do not want to. It's your PR and it's on you to make it ready for merge.